rex_xai/responsibility.py

```diff
@@ -116,12 +116,21 @@ def causal_explanation(
         np.random.seed(args.seed + process)
         tt.manual_seed(args.seed + process)
 
-    search_tree = initialise_tree(
-        data.model_height,
-        data.model_width,
-        args.distribution,
-        args.distribution_args,
-    )
+    if args.mode == "voxel":
+        search_tree = initialise_tree(
+            data.model_height,
+            data.model_width,
+            args.distribution,
+            args.distribution_args,
+            d_lim=data.model_depth,
+        )
+    else:
+        search_tree = initialise_tree(
+            data.model_height,
+            data.model_width,
+            args.distribution,
+            args.distribution_args,
+        )
```

Comment:
This can be simplified:

```suggestion
    search_tree = initialise_tree(
        data.model_height,
        data.model_width,
        args.distribution,
        args.distribution_args,
        d_lim=data.model_depth,
    )
```

If `data.mode` is not voxel, then `data.model_depth` will be `None`. Providing `d_lim=None` to `initialise_tree` will give the correct behaviour for non-voxel data, so the `if` statement here isn't needed.

This behaviour is already covered by your existing tests! :)